### PR TITLE
added automatic registering of model to ai solution

### PR DESCRIPTION
**Pull request by IanAtCredo**
## Change description

Added automatic association of model with ai solution id. Also made config file simpler.

### CODE REVIEW

File: credoai/utils/credo_api_utils.py [hunk 1]
```diff
@@ -7,25 +7,23 @@
 from dotenv import dotenv_values
 from json_api_doc import deserialize, serialize
 
+CREDO_URL = "https://api.credo-qa.com"
```
Comment (credo-eli) [inline credoai/utils/credo_api_utils.py:10]: The default prod URL is api.credo.ai, e.g. https://api.credo.ai/api/v1/credoai/me/setting

Credo AI employees would override to https://api.credo-qa.com for dogfooding on QA env

File: credoai/utils/credo_api_utils.py [hunk 2]
```diff
@@ -7,25 +7,23 @@
...
+        config = {'API_KEY': 'empty'}
     else:
         config = dotenv_values(config_file)
-    config['API_URL'] = os.path.join(config['CREDO_URL'], "api/v1/credoai")
+    config['API_URL'] = os.path.join(CREDO_URL, "api/v1/credoai")
     return config
 
 def exchange_token():
-    data = {"api_token": CONFIG['API_KEY'], "tenant": CONFIG['TENANT']}
+    data = {"api_token": CONFIG['API_KEY'], "tenant": 'credoai'}
```
Comment (credo-eli) [inline credoai/utils/credo_api_utils.py:24]: The tenant is only credoai for Credo employees. @emoralesb05 - you mentioned that it is not actually necessary to provide this?

Comment (IanAtCredo) [reply credoai/utils/credo_api_utils.py:24]: I could not get an access token without it. Something is weird

[the remark this answers is not in the document]
Comment (emoralesb05) [reply credoai/utils/credo_api_utils.py:24]: I think we need to change this from the API .. there shouldnt be a need to use `tenant` to get a JWT .. the openAPI flow basically is about pure `auth` separate from tenant API calls.. for instance the UI calls `/token` with a state key to retrieve the JWT and then `tenant` is retrieve from its claims not. 

@aproxacs how could we do something similar in lens?

[the remark this answers is not in the document]
Comment (IanAtCredo) [reply credoai/utils/credo_api_utils.py:24]: Ok, I'm going to ignore that in this PR. @aproxacs , let me know your thoughts.

Comment (emoralesb05) [reply credoai/utils/credo_api_utils.py]: The UI gets the tenant from the JWT claims calling `/token` to get the JWT. I pinged @aproxacs on the other comments to see if we are missing something.

[the remark this answers is not in the document]
Comment (aproxacs) [reply credoai/utils/credo_api_utils.py]: Tenant is required in this case. API_KEY is just a random string and does not holds `tenant` information. 

On auth0 side, we are injecting tenant to the JWT token. We are able to know tenant from organization id. 

In order to eliminate tenant here, BE need a mapping table to api_key and tenant in the global DB area, but this is not secure. So, I think using tenant here is better.

[the remark this answers is not in the document]
Comment (emoralesb05) [reply credoai/utils/credo_api_utils.py]: Not sure i agree.. we are reducing engineering complexity to increase complexity in the user side .. def would wanna discuss this a bit more before saying this is "better"

[the remark this answers is not in the document]
Comment (aproxacs) [reply credoai/utils/credo_api_utils.py]: Sure, we can discuss it. I am not talking about complexity, I am worrying about security. We are saving tenant data in each tenant DB(actually postgres schema) This is our policy in saving data. I am worrying about breaking this rule.

**Comment by IanAtCredo** [comment]
@credo-eli, this fixes the figure upload bug as well.

**Review by credo-eli — APPROVED** [review]

**Pull request merged**